src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimePropertyInfo.cs

```diff
@@ -44,10 +44,63 @@ internal RuntimePropertyInfo(
 
             scope.GetPropertyProps(tkProperty, out m_utf8name, out m_flags, out _);
 
-            Associates.AssignAssociates(scope, tkProperty, declaredType, reflectedTypeCache.GetRuntimeType(),
+            RuntimeType reflectedRuntimeType = reflectedTypeCache.GetRuntimeType();
+            Associates.AssignAssociates(scope, tkProperty, declaredType, reflectedRuntimeType,
                 out _, out _, out _,
                 out m_getterMethod, out m_setterMethod, out m_otherMethod,
                 out isPrivate, out m_bindingFlags);
+
+            // lookup getter/setter when getter and setter are inherited from base class but just a setter/getter is overridden on a sub type
+            if ((m_getterMethod != null && m_getterMethod.IsVirtual && m_setterMethod == null)
```

Comment:
nit: as mentioned earlier, `is null` and `is not null` is preferred over `== null` and `!= null` since it will prevent a call to the `==` or `!=` operator if that is overridden. In this case, `RuntimeMethodInfo` doesn't overload those operators so it doesn't matter but it is just more obvious to understand intent when using `is`.

Reply:
> RuntimeMethodInfo doesn't overload those operators so it doesn't matter

MethodInfo does overload those operators. MethodInfo operators are going to be selected to compare RuntimeMethodInfo [quick test](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAIAYAEGCMAdAEoCmAZiqQMYJQD2MA3ALABQ7EwAzggE4RaeAEw5iAVxh0wpALKkEAC3oATAJIxy9HCBzylqjVvYBvdjgt4AzHnwA2HMHr0UOACqleACglSoM/WV1TW0wAEpzSzM2S1i8AHYcMBwAXhScGHEUFFYYywBfdnygA=)

Reply:
Oops thanks; updated comment